utils/manifest-utils/src/manifest.rs

```diff
@@ -173,28 +183,55 @@ pub struct PayloadPlatform {
     pub os_version: Option<String>,
 }
 
+/// # Computation Manifest
+/// Computation Manifests lets Requestors to define a certain set of allowed actions, 
+/// to be negotiated with and approved by a Provider. 
+/// Requestors' actions will be verified against the Manifest during computation.
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct CompManifest {
+    /// # Version
+    /// Specifies a version (Semantic Versioning 2.0 specification) of the manifest.
+    #[cfg_attr(feature = "schema", schemars(with = "String", description = "Semver"))]
     pub version: Version,
+    /// # Script
+    /// Defines a set of allowed ExeScript commands and applies constraints to their arguments.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub script: Option<Script>,
+    /// # Net
+    /// Applies constraints to networking. Currently, outgoing requests to the public Internet network are covered.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub net: Option<Net>,
 }
 
+/// # Script
+/// Defines a set of allowed ExeScript commands and applies constraints to their arguments.
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Script {
+    /// Specifies a curated list of commands.
     pub commands: Vec<Command>,
+    /// Selects a default way of comparing command arguments stated in the manifest 
+    /// and the ones received in the ExeScript, 
+    /// unless stated otherwise in a command JSON object.
     #[serde(rename = "match", default)]
     pub arg_match: ArgMatch,
 }
 
+/// # Command
+#[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[derive(Clone, Debug, Serialize)]
 #[serde(untagged)]
 pub enum Command {
+    /// UTF-8 encoded string.
+    /// No command context or matching mode need to be specified.
+    /// E.g. ["run /bin/cat /etc/motd", "run /bin/date -R"]
     String(String),
+    /// UTF-8 encoded JSON string.
+    /// Command context (e.g. env) or argument matching mode need to be specified for a command.
+    /// E.g. ["{\"run\": { \"args\": \"/bin/date -R\", \"env\": { \"MYVAR\": \"42\", \"match\": \"strict\" }}}"]
     Json(Value),
```

Comment:
To let schema verification catch more errors this `Json(Value)` should be changed into a `Deserializable` type.